addons/Text_Selection/src/presenter.js

```diff
@@ -166,124 +166,108 @@ function AddonText_Selection_create() {
         }
     };
 
-    presenter.startSelection = function (et) {
-        first = parseInt($(et).attr('number'), 10);
-        if (isNaN(first)) first = parseInt($(et).attr('left'), 10);
-        if (isNaN(first)) first = parseInt($(et).closest('.selectable').attr('number'), 10);
-        if (isNaN(first)) {
-            $(et).nextAll('div').each(function () {
-                first = parseInt($(this).children('span.selectable').attr('number'), 10);
-                if (!isNaN(first)) {
-                    beforeActive = true;
-                    return false;
-                }
-            });
+    presenter.getSpanByNumber = function (number) {
+        return presenter.$view.find('.text_selection').find(`span[number='${number}']`);
+    };
+
+    presenter.handleSingleSelection = function () {
+        if (presenter.configuration.selection_type === 'SINGLESELECT') {
```

Comment:
Na przyszłość zalecałbym, aby dla takiego elementu jak ten zrobić metodę np.
```
function isSingleSelectSelectionType() {
    return presenter.configuration.selection_type === 'SINGLESELECT';
}
```
Z dwóch powodów:
* Unikasz duplikacji
* Jak nie teraz to, może w przyszłości zajdzie potrzeba na sprawdzenie typu (w niektórych addonach typ jest często sprawdzany)

Reply:
Jednak obecnie nie jest to konieczne